rest/handler.go

```diff
@@ -290,14 +288,18 @@ func (h *handler) validateAndWriteHeaders(method handlerMethod, accessPermission
 		}
 	}()
 
+	h.requestBody = NewReaderCounter(h.rq.Body)
+
```

Comment:
Is this required for the `default`/ error handling case below? If so, I think we need to consider removing the `case "":` one.